tests/test_main.py

```diff
@@ -0,0 +1,318 @@
+                           env={'MOODLE_URL': TEST_MOODLE_URL,
+                                'MOODLE_TOKEN': TEST_MOODLE_TOKEN})
+    assert result.exit_code == 0
+
+
+def test_enrol_bulk(requests_mock, tmp_path):
+    runner = CliRunner()
+    with runner.isolated_filesystem(temp_dir=tmp_path):
+        requests_mock.post(f'{TEST_MOODLE_URL}{MOODLE_WEBSERVICE_PATH}',
+                           json=post_matching_helper)
+        requests_mock.get(f'{TEST_MOODLE_URL}{MOODLE_WEBSERVICE_PATH}',
+                          json=get_matching_helper)
+
+        f = open('students.csv', 'w')
+        writer = csv.DictWriter(
+                f,
+                utils.enrol_bulk_input_csv_fieldnames()
+            )
+        writer.writeheader()
+        writer.writerows([{utils.CSV_USER_FNAME: 'Tom',
+                           utils.CSV_USER_LNAME: 'Michaels',
+                           utils.CSV_USER_EMAIL: 'tommichaels@gmail.com',
+                           utils.CSV_USER_AUTH: 'manual'},
+                          {utils.CSV_USER_FNAME: 'Phil',
+                           utils.CSV_USER_LNAME: 'Thomas',
+                           utils.CSV_USER_EMAIL: 'philthomas@gmail.com',
+                           utils.CSV_USER_AUTH: 'manual'}])
+        f.close()
+
+        # enrol-bulk <COURSE_ID> <ROLE_SHORTNAME> <USERDATA_CSV>
+        result = runner.invoke(cli, ['enrol-bulk',
+                                     '3', 'student', 'students.csv'],
+                               env={'MOODLE_URL': TEST_MOODLE_URL,
+                                    'MOODLE_TOKEN': TEST_MOODLE_TOKEN})
+        assert result.exit_code == 0
+
+
+def test_enrol_bulk_csv(requests_mock, tmp_path):
+    test_json = {'foo': 'bar'}
+    runner = CliRunner()
+    with runner.isolated_filesystem(temp_dir=tmp_path):
+        requests_mock.get(f'{TEST_MOODLE_URL}{MOODLE_WEBSERVICE_PATH}',
+                          json=test_json)
+
+        # enrol-bulk-csv  <OUTPUT_CSV>
+        result = runner.invoke(cli, ['enrol-bulk-csv', 'output.csv'],
+                               env={'MOODLE_URL': TEST_MOODLE_URL,
+                                    'MOODLE_TOKEN': TEST_MOODLE_TOKEN})
+        assert result.exit_code == 0
+        assert os.stat("output.csv").st_size != 0
+
+
+def test_import_bulk_csv(requests_mock, tmp_path):
+    test_json = {'foo': 'bar'}
+    runner = CliRunner()
+    with runner.isolated_filesystem(temp_dir=tmp_path):
+        requests_mock.get(f'{TEST_MOODLE_URL}{MOODLE_WEBSERVICE_PATH}',
+                          json=test_json)
+
+        # import-bulk-csv  <OUTPUT_CSV>
+        result = runner.invoke(cli, ['import-bulk-csv', 'output.csv'],
+                               env={'MOODLE_URL': TEST_MOODLE_URL,
+                                    'MOODLE_TOKEN': TEST_MOODLE_TOKEN})
+        assert result.exit_code == 0
+        assert os.stat("output.csv").st_size != 0
+
+
+# Not sure what import_course returns
```

Comment:
Yeah, it doesn't return anything right now. It will just throw an exception if there are any failures.